### ISSUE TITLE
extractBib property in Bootstrapping ignored when input files are txt instead of pdf

### ISSUE THREAD

**Issue by bolandka** [infolis/infoLink#79]
General question: should I fix this in the Bootstrapping class or would we like to allow chaining algorithms via json calls?

I.e. call algorithm A, apply algorithm B on its output.
This way, we could, for example, allow for learning patterns and directly resolving the generated textual references without writing an extra class calling these classes one after another. 
ApplyPatternAndResolve could be used this way, too, I'm moving the Resolve part to a separate class, anyway (to allow for the learning and resolving thing and to allow for calling the Resolver separately). Might be nicer to allow chaining than to write x classes for each combination of calls.

@kba @Dommii: What do you think?

**Comment by kba** [comment]
Good point to raise, I guess, there are more instances of such questions (e.g. on-demand indexing). I think we should best discuss this soon and decide on a pattern, whether we want to delegate algorithm chaining to the user or find a scalable, reusable way for sub-executions in the algorithms.
